rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_manager.py

```diff
@@ -0,0 +1,351 @@
+    def __init__(self, config, nav_path):
+        self.config = config
+        self.fleet_name = self.config["rmf_fleet"]["name"]
+
+        super().__init__(f'{self.fleet_name}_fleet_manager')
+
+        self.robots = {}  # Map robot name to state
+        self.docks = {}  # Map dock name to waypoints
+
+        for robot_name, robot_config in self.config["robots"].items():
+            self.robots[robot_name] = State()
+        assert(len(self.robots) > 0)
+
+        profile = traits.Profile(geometry.make_final_convex_circle(
+            self.config['rmf_fleet']['profile']['footprint']),
+            geometry.make_final_convex_circle(
+                self.config['rmf_fleet']['profile']['vicinity']))
+        self.vehicle_traits = traits.VehicleTraits(
+            linear=traits.Limits(
+                *self.config['rmf_fleet']['limits']['linear']),
+            angular=traits.Limits(
+                *self.config['rmf_fleet']['limits']['angular']),
+            profile=profile)
+        self.vehicle_traits.differential.reversible =\
+            self.config['rmf_fleet']['reversible']
+
+        self.create_subscription(
+            RobotState,
+            'robot_state',
+            self.robot_state_cb,
+            100)
+
+        transient_qos = QoSProfile(
+            history=History.RMW_QOS_POLICY_HISTORY_KEEP_LAST,
+            depth=1,
+            reliability=Reliability.RMW_QOS_POLICY_RELIABILITY_RELIABLE,
+            durability=Durability.RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL)
+
+        self.create_subscription(
+            DockSummary,
+            'dock_summary',
+            self.dock_summary_cb,
+            qos_profile=transient_qos)
+
+        self.path_pub = self.create_publisher(
+            PathRequest,
+            'robot_path_requests',
+            qos_profile=qos_profile_system_default)
+
+        self.mode_pub = self.create_publisher(
+            ModeRequest,
+            'robot_mode_requests',
+            qos_profile=qos_profile_system_default)
+
+        self.task_id = -1
+
+        @app.get('/open-rmf/rmf_demos_fm/status/')
+        async def position(robot_name: Optional[str] = None):
+            data = {'data': {},
+                    'success': False,
+                    'msg': ''}
+            if robot_name is None:
+                data['data']['all_robots'] = []
+                for robot_name in self.robots:
+                    state = self.robots.get(robot_name)
+                    if state is None or state.state is None:
+                        return data
+                    data['data']['all_robots'].append(
+                        self.get_robot_state(state, robot_name))
+            else:
+                state = self.robots.get(robot_name)
+                if state is None or state.state is None:
+                    return data
+                data['data'] = self.get_robot_state(state, robot_name)
+            data['success'] = True
+            return data
```

Comment:
Wanna sneak in another comment. It is preferred to specify the data model of the response msg, [example](https://fastapi.tiangolo.com/tutorial/response-model/). The data model will have the same structure as what is mentioned on the [readme](https://github.com/xiyuoh/rmf_demos/tree/feature/demo_fleet_adapter/rmf_demos_fleet_adapter#response-body-1)

This will be helpful for users to view the format of all endpoints via the swagger UI. (e.g. when running office world,  http://127.0.0.1:8001/docs). This recommendation also applies to other requests and response types of all exisitng fastapi endpoints.

Reply:
:+1: got it and done!